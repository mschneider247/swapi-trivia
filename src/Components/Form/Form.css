.form {
  align-items: center;
  display: flex;
  flex-direction: column;
  text-align: center;
}

.form-p {
  color: #FFFFFF;
  font-size: 1rem;
}

.input-fill {
  font-size: 1rem;
  text-align: center;
  margin: 10px;
  padding: 10px;
  width: 500px;
}

.input-container {
  display: flex;
}

.input-radio {
  margin: 10px 30px;
}

button {
  background-color: #EEDB00;
  border: none;
  font-size: 1rem;
  margin: 10px;
  padding: 10px;
  width: 300px;
}

button:hover {

}
