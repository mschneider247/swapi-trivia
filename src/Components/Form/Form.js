import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor(props) {
    super();
    this.state = {
      name: '',
      quote: '',
      status: ''
    }
    this.props = props;
  }

  updateFormState = (event, key) => {
    this.setState({ [key]: event.target.value})
  }

  buttonErrorHandling = () => {
    const stateValues = Object.values(this.state);
    let checkValues = true;
    stateValues.forEach(value => {
      if (value === '') {
        checkValues = false;
      }
    })
    if (checkValues === true) {
      this.props.updateUserState(this.state)
    }
  }

  render() {
    return (
      <header>
        <h1>Enter Your Information!</h1>
        <input
          type="text"
          placeholder="Name"
          name="name"
          value={this.state.name}
          onChange={event => this.updateFormState(event, 'name')} 
        />
        <input
          type="text"
          placeholder="Favorite Quote"
          name="quote"
          value={this.state.quote}
          onChange={event => this.updateFormState(event, 'quote')} 
        />
        <input
          type="text"
          placeholder="Your Status"
          name="status"
          value={this.state.status}
          onChange={event => this.updateFormState(event, 'status')} 
        />
        <button onClick={this.buttonErrorHandling}>Submit</button>
      </header>
    )
  }
}

export default Form;