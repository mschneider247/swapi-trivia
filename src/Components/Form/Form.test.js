import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Form
        updateUserState={jest.fn()}
      />);

      expect(wrapper).toMatchSnapshot();
    });

  it('should update name in form state when updateFormState is run', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<Form
        updateUserState={mockFunction}
      />);

    const inputEvent = {
      target: {
        name: 'name', value: 'Mike'
      }
    }

    wrapper.instance().updateFormState(inputEvent)

    expect(wrapper.state('name')).toEqual(inputEvent.target.value)
  })

  it('should update quote in form state when updateFormState is run', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<Form
        updateUserState={mockFunction}
      />);

    const inputEvent = {
      target: {
        name: 'quote', value: 'Hey'
      }
    }

    wrapper.instance().updateFormState(inputEvent)

    expect(wrapper.state('quote')).toEqual(inputEvent.target.value)
  })

  it('should update status in form state when updateFormState is run', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<Form
        updateUserState={mockFunction}
      />);

    const inputEvent = {
      target: {
        name: 'status', value: 'Cool'
      }
    }

    wrapper.instance().updateFormState(inputEvent)

    expect(wrapper.state('status')).toEqual(inputEvent.target.value)
  })

  // it('should call errorHandling when updateFormState is called', () => {
  //   const mockFunction = jest.fn()
  //   const wrapper = shallow(<Form
  //       updateUserState={mockFunction}
  //     />);
  //
  //   const mockEvent = {
  //     preventDefault: jest.fn()
  //   }
  //
  //   const mockErrorHandling = jest.fn()
  //
  //   wrapper.instance().submitSearch(mockEvent);
  //
  //   expect(wrapper.instance().errorHandling).toHaveBeenCalled();
  //
  // })
});
