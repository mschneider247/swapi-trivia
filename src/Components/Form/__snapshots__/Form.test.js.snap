// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<main
  className="form"
>
  <p
    className="form-p"
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    Please enter values for each input!
  </p>
  <input
    className="input-fill"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="input-fill"
    name="quote"
    onChange={[Function]}
    placeholder="Favorite Quote"
    type="text"
    value=""
  />
  <div
    className="input-container"
  >
    <label
      className="input-radio"
    >
      <input
        name="status"
        onChange={[Function]}
        type="radio"
        value="PADAWAN"
      />
       PADAWAN
      <span
        className="checkmark"
      />
    </label>
    <label
      className="input-radio"
    >
      <input
        name="status"
        onChange={[Function]}
        type="radio"
        value="JEDI"
      />
       JEDI
      <span
        className="checkmark"
      />
    </label>
    <label
      className="input-radio"
    >
      <input
        name="status"
        onChange={[Function]}
        type="radio"
        value="YODA"
      />
       YODA
      <span
        className="checkmark"
      />
    </label>
  </div>
  <Link
    to="/"
  >
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </Link>
</main>
`;
